kotest-assertions-arrow-fx-coroutines/src/commonMain/kotlin/io/kotest/assertions/arrow/fx/coroutines/Resource.kt

```diff
@@ -0,0 +1,141 @@
+package io.kotest.assertions.arrow.fx.coroutines
+
+import arrow.continuations.generic.AtomicRef
+import arrow.continuations.generic.update
+import arrow.core.NonEmptyList
+import arrow.core.ValidatedNel
+import arrow.core.identity
+import arrow.core.invalidNel
+import arrow.core.traverseValidated
+import arrow.core.valid
+import arrow.fx.coroutines.ExitCase
+import io.kotest.core.TestConfiguration
+import arrow.fx.coroutines.Platform
+import arrow.fx.coroutines.Resource
+import arrow.fx.coroutines.bracketCase
+import io.kotest.common.runBlocking
+import io.kotest.core.listeners.TestListener
+import io.kotest.core.spec.Spec
+
+public suspend fun <A> Resource<A>.shouldBeResource(a: A): A =
+  use { it shouldBe a }
+
+public suspend fun <A> Resource<A>.shouldBeResource(expected: Resource<A>): A =
+  zip(expected).use { (a, b) -> a shouldBe b }
+
+/**
+ * ensures that all finalizers of [resource] are invoked for the given [TestConfiguration],
+ * by registering a listener.
+ *
+ * ```kotlin
+ * interface Database { // given a Dependency
+ *   fun isRunning(): Boolean
+ *   suspend fun version(): String?
+ * }
+ *
+ * // and a Way to create it, here with Hikari and SqlDelight
+ * fun database(hikari: HikariDataSource): Database =
+ *   object : Database {
+ *     override fun isRunning(): Boolean = hikari.isRunning
+ *     override suspend fun version(): String? =
+ *       hikari.queryOneOrNull("SHOW server_version;") { string() }
+ *    }
+ *
+ * fun hikari(): Resource<HikariDataSource> =
+ *   Resource.fromCloseable {
+ *     HikariDataSource(
+ *      HikariConfig().apply {
+ *       // config settings
+ *      }
+ *     )
+ *   }
+ * fun sqlDelight(hikariDataSource: HikariDataSource): Resource<SqlDelight> =
+ *   Resource.fromCloseable { hikariDataSource.asJdbcDriver() }.map { driver -> SqlDelight(driver) }
+ *
+ * // we can define a Database Resource
+ * fun database(): Resource<Database> =
+ *  hikari().flatMap { hikari ->
+ *   sqlDelight(hikari).map { sqlDelight -> database(hikari) }
+ *  }
+ *
+ * // and define Tests with our backend Application, here with Ktor
+ * class HealthCheckSpec :
+ *   StringSpec({
+ *     val database: Database = resource(database())
+ *
+ *     "healthy" {
+ *       testApplication {
+ *         application { app(database) }
+ *         val response = client.get("/health")
+ *         response.status shouldBe HttpStatusCode.OK
+ *         response.bodyAsText() shouldBe Json.encodeToString(HealthCheck("14.1"))
+ *       }
+ *     }
+ *   })
+ * ```
+ */
+public fun <A> TestConfiguration.resource(resource: Resource<A>): A =
+  runBlocking { TestResource(resource).also(this::listener).value() }
```

Comment:
Hi @sksamuel is there a way here to remove `runBlocking`, but still call `value()` in `TestResource`.
The other option is changing the `resource` function to be suspended, but it defeats the purpose of having this function in the first place if one has to invoke it in each Test
cc @nomisRev